examples/gno.land/r/gov/dao/v3/impl/prop_requests.gno

```diff
@@ -0,0 +1,216 @@
+		Description: "This prpoposal is looking to change the actual govDAO Law",
+		Callback:    changeLawCallback,
+		Metadata: &ChangeLawMetadata{
+			Law: law,
+		},
+	}
+}
+
+type AddMemberMetadata struct {
+	Member    *memberstore.Member
+	Portfolio string
+	Tier      string
+	Addr      std.Address
+	Caller    *memberstore.Member
+}
+
+func (a *AddMemberMetadata) IsMetadata() {}
+func (a *AddMemberMetadata) String() string {
+	return ufmt.Sprintf("A new member with address %v is proposed to be on tier %v. Provided Portfolio information:\n\n%v", a.Addr, a.Tier, a.Portfolio)
+}
+
+func addMemberCallback(m proxy.Metadata) error {
+	amm, ok := m.(*AddMemberMetadata)
+	if !ok {
+		panic("metadata type for adding a member is wrong")
+	}
+
+	amm.Caller.RemoveInvitationPoint()
+
+	return memberstore.Get().SetMember(amm.Tier, amm.Addr, amm.Member)
+}
+
+func NewAddMemberRequest(addr std.Address, tier string, portfolio string) proxy.ProposalRequest {
+	_, ok := memberstore.Tiers.GetTier(tier)
+	if !ok {
+		panic("provided tier does not exists")
+	}
+
+	if tier != memberstore.T1 && tier != memberstore.T2 {
+		panic("Only T1 and T2 members must be added by proposal. To add a T3 member use AddMember function directly.")
+	}
+
+	if portfolio == "" {
+		panic("A portfolio for the proposed member is required")
+	}
+
+	member, _ := memberstore.Get().GetMember(std.OriginCaller())
+	if member == nil {
+		panic("proposer is not a member")
+	}
+
+	if member.InvitationPoints <= 0 {
+		panic("proposer does not have enough invitation points for inviting new people to the board")
+	}
+
+	return proxy.ProposalRequest{
+		Title:       "New Member porposal",
+		Description: "This prpoposal is looking to add a new member to the board.",
+		Callback:    addMemberCallback,
+		Metadata: &AddMemberMetadata{
+			Member:    memberByTier(tier),
+			Portfolio: portfolio,
+			Tier:      tier,
+			Addr:      addr,
+			Caller:    member,
+		},
+	}
+}
+
+type RemoveMemberMetadata struct {
+	Addr   std.Address
+	Tier   string
+	Reason string
+}
+
+func (a *RemoveMemberMetadata) IsMetadata() {}
+func (a *RemoveMemberMetadata) String() string {
+	return ufmt.Sprintf("Member with address %v will be withdrawn.\n\n REASON: %v.", a.Addr, a.Reason)
+}
+
+func removeMemberCallback(m proxy.Metadata) error {
+	rmm, ok := m.(*RemoveMemberMetadata)
+	if !ok {
+		panic("metadata type for removing a member is wrong")
+	}
+
+	memberstore.Get().RemoveMember(rmm.Addr)
+
+	return nil
+}
+
+func NewWithdrawMemberRequest(addr std.Address, reason string) proxy.ProposalRequest {
+	member, tier := memberstore.Get().GetMember(addr)
+	if member == nil {
+		panic("user we want to remove not found")
+	}
+
+	if tier == memberstore.T1 && reason == "" {
+		panic("T1 user removals must contains a reason.")
+	}
+
+	return proxy.ProposalRequest{
+		Title:       "Member Withdrawn",
+		Description: "This prpoposal is looking to remove a member from the board.",
+		Callback:    removeMemberCallback,
```

Comment:
what prevent me to do this:

```go
proxy.NewWithdrawMemberRequest("g12345", "foo").Callback()
```

We should unexport Callback.

Reply:
Calls to member storage are protected, and can only be done by the actual govDAO implementation, so that specific call will fail. In any case, I'll make callback private to avoid confusion in other implementations.

Reply:
Callback was moved to private and added the ProposalRequestBuilder struct to generate the Proposal requests without being able to access the internal fields.